pkg/api/bzz_test.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package api_test
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"mime"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/ethersphere/bee/pkg/api"
+	"github.com/ethersphere/bee/pkg/collection/entry"
+	"github.com/ethersphere/bee/pkg/file"
+	"github.com/ethersphere/bee/pkg/file/splitter"
+	"github.com/ethersphere/bee/pkg/jsonhttp"
+	"github.com/ethersphere/bee/pkg/jsonhttp/jsonhttptest"
+	"github.com/ethersphere/bee/pkg/logging"
+	"github.com/ethersphere/bee/pkg/manifest/jsonmanifest"
+	smock "github.com/ethersphere/bee/pkg/storage/mock"
+	"github.com/ethersphere/bee/pkg/swarm"
+	"github.com/ethersphere/bee/pkg/tags"
+)
+
+func TestBzz(t *testing.T) {
+	var (
+		bzzDownloadResource = func(addr, path string) string { return "/bzz:/" + addr + "/" + path }
+		storer              = smock.NewStorer()
+		sp                  = splitter.NewSimpleSplitter(storer)
+		client              = newTestServer(t, testServerOptions{
+			Storer:         storer,
+			ManifestParser: jsonmanifest.NewParser(),
+			Tags:           tags.NewTags(),
+			Logger:         logging.New(ioutil.Discard, 5),
+		})
+	)
+
+	t.Run("download-file-by-path", func(t *testing.T) {
+		fileName := "sample.html"
+		filePath := "test/" + fileName
+		missingFilePath := "test/missing"
+		sampleHtml := `<!DOCTYPE html>
+		<html>
+		<body>
+
+		<h1>My First Heading</h1>
+
+		<p>My first paragraph.</p>
+
+		</body>
+		</html>`
+
+		var err error
+		var fileReference swarm.Address
+		var manifestFileReference swarm.Address
```

Comment:
I would maybe put this in the `TestBzz` scope and just have these individual tests below. Don't see a need for sub-tests.